activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -162,14 +164,19 @@ def visit_Arel_Nodes_SelectCore(o, collector)
           collect_nodes_for o.havings, collector, " HAVING ", " AND "
           collect_nodes_for o.windows, collector, " WINDOW "
 
-          collector
+          maybe_visit o.comment, collector
         end
 
         def visit_Arel_Nodes_OptimizerHints(o, collector)
           hints = o.expr.map { |v| sanitize_as_sql_comment(v) }.join(" ")
           collector << "/*+ #{hints} */"
         end
 
+        def visit_Arel_Nodes_Comment(o, collector)
+          comments = o.values.map { |v| sanitize_as_sql_comment(v) }.join(" ")
```

Comment:
How about joining multiple comments with `"; "`?

I think it is one solution to mitigate the issue `Post.annotate("post comment").joins(:tags).merge(Tag.annotate("tag comment")) #=> "/* post comment tag comment */"`.

Reply:
I don't think text content should be inserted into the user's annotation if it can be avoided. As a consumer of the API I wouldn't want that.

After more reflection I think the best approach by far is actually to wrap each annotation in its own delimiters, e.g.,

```
Post.annotate("post comment").joins(:tags).merge(Tag.annotate("tag comment"))
#=> "... /* post comment */ /* tag comment */"
``` 

This solves the concerns with running them together, it makes it much simpler to detect or parse out annotation in instrumentation, and it prevents different developers from stepping on one another's annotations.

I've been using a "backport" of this API in an experimental branch for a large refactor and after working with it a bit this is the behavior I'd want to have.